skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,367 @@
+        Converted skpro distribution object.
+        """
+        from skpro.distributions.laplace import Laplace
+        from skpro.distributions.lognormal import LogNormal
+        from skpro.distributions.normal import Normal
+        from skpro.distributions.poisson import Poisson
+        from skpro.distributions.t import TDistribution
+
+        ngboost_dists = {
+            "Normal": Normal,
+            "Laplace": Laplace,
+            "TDistribution": TDistribution,
+            "Poisson": Poisson,
+            "LogNormal": LogNormal,
+        }
+
+        skpro_dist = None
+
+        mu_sigma = ["Normal", "TDistribution", "LogNormal"]
+        mu_scale = ["Laplace"]
+        mu_only = ["Poisson"]
+
+        if self.dist in ngboost_dists and self.dist in mu_sigma:
+            skpro_dist = ngboost_dists[self.dist](
+                mu=pred_mean, sigma=pred_std, index=index, columns=columns
+            )
+
+        elif self.dist in ngboost_dists and self.dist in mu_scale:
+            skpro_dist = ngboost_dists[self.dist](
+                mu=pred_mean, scale=pred_std, index=index, columns=columns
+            )
+
+        elif self.dist in ngboost_dists and self.dist in mu_only:
+            skpro_dist = ngboost_dists[self.dist](
+                mu=pred_mean, index=index, columns=columns
+            )
+
+        return skpro_dist
+
+    def _predict_proba(self, X):
+        """Predict distribution over labels for data from features.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : skpro BaseDistribution, same length as `X`
+            labels predicted for `X`
+        """
+        X = self._check_X(X)
+
+        # The returned values of the Distributions from NGBoost
+        # are different. So based on that they are split into these
+        # categories of loc,scale,mu and s.
+        # Distribution type | Parameters
+        # ------------------|-----------
+        # Normal            | loc = mean, scale = standard deviation
+        # TDistribution     | loc = mean, scale = standard deviation
+        # Poisson           | mu = mean
+        # LogNormal         | s = standard deviation, scale = exp(mean)
+        #                   |     (see scipy.stats.lognorm)
+        # Laplace           | loc = mean, scale = scale parameter
+
+        loc_scale_params = ["Normal", "Laplace", "TDistribution"]
+        mu_params = ["Poisson"]
+        s_scale = ["LogNormal"]
+        pred_mean = None
+        pred_std = None
+        # Convert NGBoost distribution to skpro BaseDistribution
+        if self.dist in loc_scale_params:
+            pred_mean = self._pred_dist(X).params["loc"]
```

Comment:
this is not a bug, but confusing naming - the two parameters passed are not always mean or std!

You do pass these into the distributions correctly though, so there is nothing to fix except the naming.